fuse_loss/include/fuse_loss/arctan_loss.h

```diff
@@ -0,0 +1,129 @@
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef FUSE_LOSS_ARCTAN_LOSS_H
+#define FUSE_LOSS_ARCTAN_LOSS_H
+
+#include <fuse_core/loss.h>
+
+#include <string>
+
+
+namespace fuse_loss
+{
+
+/**
+ * @brief The ArctanLoss loss function.
+ *
+ * This class encapsulates the ceres::ArctanLoss class, adding the ability to serialize it and load it dynamically.
+ *
+ * See the Ceres documentation for more details. http://ceres-solver.org/nnls_modeling.html#lossfunction
+ */
+class ArctanLoss : public fuse_core::Loss
+{
+public:
+  FUSE_LOSS_DEFINITIONS(ArctanLoss);
+
+  /**
+   * @brief Constructor
+   *
+   * @param[in] a ArctanLoss parameter 'a'. See Ceres documentation for more details
+   */
+  explicit ArctanLoss(const double a = 1.0);
+
+  /**
+   * @brief Destructor
+   */
+  ~ArctanLoss() override = default;
+
+  /**
+   * @brief Perform any required post-construction initialization, such as reading from the parameter server.
+   *
+   * This will be called on each plugin after construction.
+   *
+   * @param[in] name A unique name to initialize this plugin instance, such as from the parameter server.
+   */
+  void initialize(const std::string& name) override;
+
+  /**
+   * @brief Print a human-readable description of the loss function to the provided stream.
+   *
+   * @param[out] stream The stream to write to. Defaults to stdout.
+   */
+  void print(std::ostream& stream = std::cout) const override;
+
+  /**
+   * @brief Parameter 'a' accessor.
+   *
+   * @return Parameter 'a'.
+   */
+  double a() const
+  {
+    return a_;
+  }
+
+  /**
+   * @brief Parameter 'a' mutator.
+   *
+   * @param[in] a Parameter 'a'.
+   */
+  void a(const double a)
+  {
+    a_ = a;
+  }
+
+private:
+  double a_{ 1.0 };  //<! ArctanLoss parameter 'a'. See Ceres documentation for more details
+
+  // Allow Boost Serialization access to private methods
+  friend class boost::serialization::access;
+
+  /**
+   * @brief The Boost Serialize method that serializes all of the data members in to/out of the archive
+   *
+   * @param[in/out] archive - The archive object that holds the serialized class members
+   * @param[in] version - The version of the archive being read/written. Generally unused.
+   */
+  template<class Archive>
+  void serialize(Archive& archive, const unsigned int /* version */)
+  {
+    archive & boost::serialization::base_object<fuse_core::Loss>(*this);
+    archive & a_;
+  }
+};
+
+}  // namespace fuse_loss
+
+BOOST_CLASS_EXPORT_KEY(fuse_loss::ArctanLoss);
```

Comment:
This macro is obviously included somewhere, but the "include what you use" principle would recommend that the boost header be included here as well. Similarly, the pluginlib macro header should be included in the cpp files.

Reply:
Done. Also moved the pluginlib header up.